zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -181,4 +154,37 @@ public void updateDomainBasedAuthInfo(ServerCnxn cnxn) {
       }
     }
   }
+
+  /**
+   * The watcher used to listen on client uri - domain mapping root path for mapping update
+   * Extends DumbWatcher instead of ServerCnxn here some methods in ServerCnxn are not accessible from here
+   */
+  public class MappingRootWatcher extends DumbWatcher {
+    @Override
+    public void process(WatchedEvent event) {
+      LOG.info("Processing watched event: {}", event.toString());
+      parseZNodeMapping();
+      // Update AuthInfo for all the known connections.
+      // Note : It is not ideal to iterate over all plaintext connections which are connected over non-TLS but right now
+      // there is no way to find out if connection on unified port is using SSLHandler or nonSSLHandler. Anyways, we
+      // should not ideally have any nonSSLHandler connections on unified port after complete rollout.
```

Comment:
Checkout `ServerCnxn.isSecure()` and `ServerCnxn.getClientCertificateChain()`. These could be ways to detect TLS connections over unified port.

Reply:
Since this is not LinkedIn specific codebase, let's not worry about TLS rollout or shutting off plaintext port. Let's do what's right in a more general sense.

Reply:
Thanks! These are good suggestions. This part of code is not modified in the PR, and is not related to the issue we are trying to fix here, so if we want to change them, we will need another PR.